Content.Server/Weapons/Melee/MeleeWeaponSystem.cs

```diff
@@ -288,4 +289,34 @@ private void OnChemicalInjectorHit(EntityUid owner, MeleeChemicalInjectorCompone
             _bloodstream.TryAddToChemicals(ent, individualInjection, bloodstream);
         }
     }
+    private void OnChemicalPickerHit(EntityUid owner, MeleeChemicalPickerComponent comp, MeleeHitEvent args)
+    {
+        if (!args.IsHit ||
+            !args.HitEntities.Any() ||
+            !_solutions.TryGetSolution(owner, comp.Solution, out var solutionContainer))
+        {
+            return;
+        }
+
+        var hitBloodstreams = new List<(EntityUid Entity, BloodstreamComponent Component)>();
+        var bloodQuery = GetEntityQuery<BloodstreamComponent>();
+
+        foreach (var entity in args.HitEntities)
+        {
+            if (Deleted(entity))
+                continue;
+
+            if (bloodQuery.TryGetComponent(entity, out var bloodstream))
+                hitBloodstreams.Add((entity, bloodstream));
+        }
+
+        if (!hitBloodstreams.Any())
+            return;
+
+        var transferAmount = (comp.TransferAmount * comp.TransferEfficiency) / hitBloodstreams.Count;
+        foreach (var (ent, bloodstream) in hitBloodstreams)
+        {
+            var result = _solutions.TryTransferSolution(owner, solutionContainer, bloodstream.BloodSolution, transferAmount);
+        }
```

Comment:
you could easily condense this to remove the hitBloodstreams list and make the logic more clear. Just do the transfer call in the first loop instead of doing a second one later.